benchmark/single-source/Flatten.swift

```diff
@@ -0,0 +1,962 @@
+
+    var f: [Int] = []
+
+    for array in input {
+      for element in array {
+        f.append(element)
+      }
+    }
+
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastElement))
+  }
+}
+
+@inline(never)
+public func run_FlattenAnySeqSeqArray4_forinBatch(_ N: Int) {
+  for _ in 1...N {
+    let input = AnySequence(sequenceArray4)
+
+    var f: [Int] = []
+
+    for array in input {
+      f += array
+    }
+
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastElement))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorVal_flatMap(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.flatMap { [$0.a, $0.r, $0.g, $0.b] }
+
+    CheckResults((f.count, f.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorVal_map_joined(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.map({ [$0.a, $0.r, $0.g, $0.b] }).joined()
+
+    CheckResults((f.count, f.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorVal_lazy_map_joined(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.lazy.map({ [$0.a, $0.r, $0.g, $0.b] }).joined()
+
+    let cl = f.reduce(into: CountLast(), CountLast.collect)
+    CheckResults((cl.count, cl.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorVal_forinNaive(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    var f: [UInt8] = []
+
+    for c in input {
+      f.append(c.a)
+      f.append(c.r)
+      f.append(c.g)
+      f.append(c.b)
+    }
+
+    CheckResults((f.count, f.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorValUnsafe(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.withUnsafeBufferPointer {
+      $0.baseAddress!.withMemoryRebound(to: UInt32.self, capacity: input.count) {
+        UnsafeBufferPointer(start:$0, count: input.count).flatMap {
+          (rawColor: UInt32) -> [UInt8] in
+          let rgba = rawColor.bigEndian
+          let argb = (rgba << 24) | (rgba >> 8) // rotate
+          return stride(from:24, through:0, by: -8).map {
+            UInt8(truncatingIfNeeded: argb >> UInt32($0))
+          }
+        }
+      }
+    }
+    // print(f.suffix(4))
+
+    CheckResults((f.count, f.last!) == (flatSize, lastBlue))
+  }
+}
+
+@inline(never)
+public func run_FlattenArrayColorValUnsafe2(_ N: Int) {
+  for _ in 1...N {
+    let input = arrayColorVals
+
+    let f = input.withUnsafeBufferPointer {
+      $0.baseAddress!.withMemoryRebound(to: UInt32.self, capacity: input.count) {
+        UnsafeBufferPointer(start:$0, count: input.count).map { rgba in
+          // rotate (inverted direction, because x86 is little endian)
```

Comment:
I'm not sure why you want x86 specific benchmark logic. You could view the memory as `UInt8` or just not include this nonprotable case.